### PR TITLE
Rev History Plugin

### PR DESCRIPTION
**Pull request by codemis**
I created a plugin for getting the revision history using a GET API request.  Please see the README for further details.

### CODE REVIEW

File: lib/plugins/revhistory/_test/plugin_revhistory.test.php [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+        $this->plugin->testing = true;
+        $this->eventHandler = $this->getMockBuilder('Doku_Event')
+            ->setConstructorArgs(array('ACTION_ACT_PREPROCESS', 'revhistory'))
+            ->getMock();
+        /**
+         * Overwrite the config directories so it uses our test directories.
+         * For some reason when DokuWiki moves the test files, they do not get moved and throw an error.
+         */
+        $conf['changelog']          = dirname(__FILE__) . '/data/meta/_dokuwiki.changes';
+        $conf['media_changelog']    = dirname(__FILE__) . '/data/meta/_media.changes';
```
Comment (phillip-hopper) [inline lib/plugins/revhistory/_test/plugin_revhistory.test.php]: The command to copy your _test/data directories is 

```
TestUtils::rcopy(TMP_DIR, dirname(__FILE__) . '/data/');
```

See https://github.com/unfoldingWord-dev/Door43/blob/development/lib/plugins/door43obs/_test/CreateNow.test.php for an example.

**Comment by phillip-hopper** [comment]
@codemis, It would help if you squashed this pull request down to just 1 or 2 commits.

File: lib/plugins/revhistory/_test/plugin_revhistory.test.php (hunk 1 above)
Comment (codemis) [inline lib/plugins/revhistory/_test/plugin_revhistory.test.php]: Thank you Phillip.  I tried that, and when I try to open the config files it throws a file not found error.  I looked in the tmp folder, and the data folder isn't there.  According to the docs, it should move the files to temp folder automatically, without rcopy command,  but it doesn't work either.   I still get a file not found error.  

Look at the Environment section in the testing doc: https://www.dokuwiki.org/devel:unittesting.  

I am on a Mac, and am sure it is a permissions issue.  Does it work on your platform?

**Comment by codemis** [comment]
@phillip-hopper : Thank you!  I actually did not know you can squash your commits into a fewer commit statements.  I found [this article](http://jamescooke.info/git-to-squash-or-not-to-squash.html) that explains it in detail.  Cool!  
Is this the preferred way for submitting pull requests to Unfolding Word?  If so,  I will be happy to rebase my code into fewer commits.  I will take care of it on Monday.  Thank you!

**Comment by richmahn** [comment]
Very useful to see the article on how to squash commits...I myself have been told to do that but never got around to learning.

File: lib/plugins/revhistory/_test/plugin_revhistory.test.php (hunk 1 above)
Comment (phillip-hopper) [inline lib/plugins/revhistory/_test/plugin_revhistory.test.php]: On my machine (Ubuntu) the test data is copied to `/tmp/dwtests-1441020121.9295/`. Of course the last part changes each time.  If you look at `_test/bootstrap.php` starting around line 23 you can see how the system determines where the test data will be copied. 

My experience is that the test data in the plugins is not copied automatically.  The documentation is misleading here - only files in the main `_test/data` directory are copied automatically.

**Comment by codemis** [comment]
@phillip-hopper Thank you for pointing me to the bootstrap file.  I changed my permissions on the temp directory, and got the TestUtils working.  Awesome!  Also, I was able to squash my commits down.  Thank you for directing me that way.

**Pull request merged**